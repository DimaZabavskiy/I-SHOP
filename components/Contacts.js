import React from 'react';
import './pages/Page_Basket.css'

class Contacts extends React.PureComponent {
    
    state = {
        name: '',
        email: '',
    }
    
    saveName = () => {
        this.props.cbSaveName();
    }
    setName = EO => {
            this.setState ({
                name: EO.target.value,
            })
    }
    setEmail = EO => {
            this.setState ({
                email: EO.target.value,
            })
    }

    render () {
        console.log('render Contacts');

        return (
            <div className='Contacts'>
                <span>Контактная информация:</span>    
                <br />
                <input type='text' onChange={this.setName} placeholder='ФИО' />
                        {!this.state.name && <span className='redSpan'>Заполнение поля обязательно!</span> }
                <br />
                <input type='email' onChange={this.setEmail} placeholder='email' />
                        {!this.state.email && <span className='redSpan'>Заполнение поля обязательно!</span> }
                <br />
                <textarea type='text' placeholder='Дополнительная информация' />
                <br />
                { (this.state.name && this.state.email) && <button onClick={this.saveName}>Оформить заказ</button>}
            </div>
        )
    }

}

export default Contacts;