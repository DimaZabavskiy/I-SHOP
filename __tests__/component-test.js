import React from 'react';
import renderer from 'react-test-renderer';

import Contacts from '../components/Contacts';

test('работа компонента Contacts', () => {

  const component = renderer.create(
    <Contacts />
  );

  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
    
  const elemFIO = component.root.find( el => el.type=='input' && el.props.placeholder == 'ФИО' ); 
  let fioEO = {
    target: { value: "FIO" }
  };
  elemFIO.props.onChange(fioEO);

  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  const elemEmail = component.root.find( el => el.type=='input' && el.props.placeholder == 'email' ); 
  let emailEO = {
    target: { value: "EMAIL" }
  };
  elemEmail.props.onChange(emailEO);

  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
/*
  saveName = () => {
    console.log('hello');
    this.setState ({
      name: '',
      email: '',
    })
  }

  const buttonElem = component.root.find( el => el.type=='button' ); 
  buttonElem.onClick = this.saveName;
  buttonElem.props.onClick();
  
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();*/

});
