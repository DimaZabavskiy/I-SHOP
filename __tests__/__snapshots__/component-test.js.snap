// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа компонента Contacts 1`] = `
<div
  className="Contacts"
>
  <span>
    Контактная информация:
  </span>
  <br />
  <input
    onChange={[Function]}
    placeholder="ФИО"
    type="text"
  />
  <span
    className="redSpan"
  >
    Заполнение поля обязательно!
  </span>
  <br />
  <input
    onChange={[Function]}
    placeholder="email"
    type="email"
  />
  <span
    className="redSpan"
  >
    Заполнение поля обязательно!
  </span>
  <br />
  <textarea
    placeholder="Дополнительная информация"
    type="text"
  />
  <br />
  
</div>
`;

exports[`работа компонента Contacts 2`] = `
<div
  className="Contacts"
>
  <span>
    Контактная информация:
  </span>
  <br />
  <input
    onChange={[Function]}
    placeholder="ФИО"
    type="text"
  />
  <br />
  <input
    onChange={[Function]}
    placeholder="email"
    type="email"
  />
  <span
    className="redSpan"
  >
    Заполнение поля обязательно!
  </span>
  <br />
  <textarea
    placeholder="Дополнительная информация"
    type="text"
  />
  <br />
  
</div>
`;

exports[`работа компонента Contacts 3`] = `
<div
  className="Contacts"
>
  <span>
    Контактная информация:
  </span>
  <br />
  <input
    onChange={[Function]}
    placeholder="ФИО"
    type="text"
  />
  <br />
  <input
    onChange={[Function]}
    placeholder="email"
    type="email"
  />
  <br />
  <textarea
    placeholder="Дополнительная информация"
    type="text"
  />
  <br />
  <button
    onClick={[Function]}
  >
    Оформить заказ
  </button>
</div>
`;
